scripts/reformat_docs.py

```diff
@@ -199,9 +204,12 @@ def replace_block(
             {f.name: f.contents for f in parse_fields(block_contents.group(1))})
 
     if header_formatter.is_block_valid(block):
-        return '%s%s\n' % (
-                block_contents.group(0),
-                header_formatter.convert(header_from_block(block)))
+        converted = header_formatter.convert(header_from_block(block))
+        if header_formatter.needs_new_header(file_contents) and converted:
+            return '%s%s' % (
+                    block_contents.group(0),
+                    header_formatter.convert(header_from_block(block)) + '\n')
```

Comment:
Unless I am misunderstanding something, can we not use `converted` here. Why do we now need the check `converted` is truthy? Surely if it is an empty string this amounts to returning `block_contents.group(0)`?